### PR TITLE
Update neighboring

### PR DESCRIPTION
**Pull request by chrisiacovella**
## Description
This relates to issue #10 .  The pair and neighborlist classes now also return the index of the pairs.  This information was available, just not returned during the calculate function (as it wasn't necessary for initial testing/dev of a monatomic system). 

Note:  I tried a version of the pairlist that basically just removed non-interacting pairs to reduce the number of atoms used in the force calculation (current code will have all possible pairs, and the energy is just multiplied by either 0 or 1, i.e., based on the mask).   Similar to the neighborlist, there was a n_max_neighbors variable that would pad things out.  For the range of system sizes tested these extra steps took more time than extra LJ computations.  I suspect for an NNP, this step to reduce extra interactions will likely  speed things up given the cost of the NNP.  This will be added in a follow-up commit to this PR; if it is ultimately not useful we can remove later.  I'm not sure if this should be a new class or just part of the same class

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [x] Make pair information accessible to allow for mixed systems
  - [x] Improve doc strings
  
 Will move the other action items to a separate PR as they are not actually necessary here and not merging will potential block progress in other items. 

>  - [ ] Add in code that reduces and pads the Pairlist 
>   - [ ] Add in example using the minimize function
>   - [ ] Create mixed LJ system to implement logic related to converting topology information into the most usable form.
>   

## Status
- [x] Ready to go

### CODE REVIEW

File: chiron/neighbors.py [hunk 1]
```diff
@@ -14,9 +14,31 @@
...
+                        f"Box vectors require distance unit, not {box_vectors.unit}"
+                    )
+                self.box_vectors = box_vectors.value_in_unit_system(unit.md_unit_system)
+            elif isinstance(box_vectors, jnp.ndarray):
+                if box_vectors.shape != (3, 3):
+                    raise ValueError(
+                        f"box_vectors should be a 3x3 array, shape provided: {box_vectors.shape}"
+                    )
+
+                self.box_vectors = box_vectors
```
Comment (wiederm) [inline chiron/neighbors.py:42]:
```suggestion
            else:
                raise ValueError(f'box_vectors need to be either `unit.Quantity` or `jnp.ndarray`')

```

Comment (wiederm) [reply chiron/neighbors.py:42]: The code lacks a default case in the `if ... elif ...` statement, which could lead to unhandled situations. Added a suggestion for such a case.

File: chiron/neighbors.py [hunk 2]
```diff
@@ -39,21 +61,10 @@ def wrap(self, xyz: jnp.array) -> jnp.array:
 
 class OrthogonalPeriodicSpace(Space):
     """
-    Calculate the periodic distance between two points.
+    Defines the simulation space for an orthogonal periodic system.
 
-    Returns
-    -------
-    Callable
```
Comment (wiederm) [inline chiron/neighbors.py:46]: Is the return type correct here?

**Review by wiederm — APPROVED** [review]
Great improvments!  This is ready to be merged.

File: chiron/neighbors.py (hunk 1 above)
Comment (chrisiacovella) [reply chiron/neighbors.py:42]: Good catch!

File: chiron/neighbors.py (hunk 2 above)
Comment (chrisiacovella) [reply chiron/neighbors.py:46]: I had initially set this up as nested functions.  I removed this when I updated the code in the PR.
